### PR TITLE
create a new blank demo app to test refacto components with minimalis…

### PR DESCRIPTION
**Pull request by rohitsharma120582**
Create a blank demo app to test the refacto components

## Technical description
* Create an ESLint configuration for TypeScript React named `@monkvision/eslint-config-typescript-react` in `configs /` directory
* Create a directory called `apps` at the root of the repository and add it to the yarn workspace
* Create a blank demo app with React & TypeScript to test our SDK named `monk-test-app`
* Use typescript-configuration and eslint-configuration from `configs/` directory

## Test the app
```
yarn
cd apps/monk-test-app
yarn start
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
 # Swap the comments on the following lines if you don't wish to use zero-installs
 # Documentation here: https://yarnpkg.com/features/zero-installs
 #!.yarn/cache
 .pnp.*
 
-node_modules
-.idea
+# modules
+node_modules/
+.docusaurus/
```
Comment (souyahia-monk) [inline .gitignore:18]: why did you add this line?

File: .gitignore [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+.idea/
+
+# testing
+coverage/
+
+# production
+build/
```
Comment (souyahia-monk) [inline .gitignore:25]: please don't do that. This line is indicating to github that we would like to ignore ALL directories called `build` in the repository. This is not a good practice and this is not what we want. If you want to ignore a specific build directory, use the gitignore file located in the NPM sub project

File: apps/monk-test-app/.eslintignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+dist
+node_modules
+.eslintrc.js
+jest.config.js
```
Comment (souyahia-monk) [inline apps/monk-test-app/.eslintignore:4]: why are you ignoring jest config ?

File: apps/monk-test-app/.eslintignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+dist
+./lib
+src/lib/**
+!src/lib/types.ts
```
Comment (souyahia-monk) [inline apps/monk-test-app/.eslintignore:8]: why are those lines here ?

File: apps/monk-test-app/package.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-utils": "^3.0.0",
+    "jest": "^29.3.1",
+    "mkdirp": "^1.0.4",
+    "prettier": "^2.7.1",
+    "regexpp": "^3.2.0",
+    "rimraf": "^3.0.2",
+    "ts-jest": "^29.0.3"
+  },
+  "scripts": {
```
Comment (souyahia-monk) [inline apps/monk-test-app/package.json:52]: could you move the "scripts" key above the dependencies please ? It's easier to read like that. Thanks!

File: apps/monk-test-app/src/index.css [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    monospace;
+}
+
+header {
+  min-height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  font-size: calc(10px + 2vmin);
+}
```
Comment (souyahia-monk) [inline apps/monk-test-app/src/index.css:21]: add a new line at the end of the file

File: configs/eslint-config-base/package.json [hunk 7]
```diff
@@ -17,9 +17,11 @@
     "eslint-config-prettier": "^8.5.0",
     "eslint-plugin-eslint-comments": "^3.2.0",
     "eslint-plugin-import": "^2.26.0",
-    "eslint-plugin-jest": "^27.1.5",
+    "eslint-plugin-jest": "^25.3.0",
```
Comment (souyahia-monk) [inline configs/eslint-config-base/package.json:20]: why did you downgrade the plugin version?

File: configs/eslint-config-base/package.json [hunk 8]
```diff
@@ -17,9 +17,11 @@
...
     "eslint-plugin-prettier": "^4.2.1",
-    "eslint-plugin-promise": "^6.1.1"
+    "eslint-plugin-promise": "^6.1.1",
```
Comment (souyahia-monk) [inline configs/eslint-config-base/package.json:22]: why did you downgrade these plugins versions?

File: configs/eslint-config-base/package.json [hunk 9]
```diff
@@ -17,9 +17,11 @@
...
+    "eslint-utils": "^3.0.0",
+    "regexpp": "^3.2.0"
```
Comment (souyahia-monk) [inline configs/eslint-config-base/package.json:24]: why did you add these dependencies in the base config?

File: configs/eslint-config-typescript-react/index.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+const OFF = 0;
+const WARN = 1;
+const ERROR = 2;
+
+module.exports = {
+  root: true,
+  extends: [
+    '@monkvision/eslint-config-typescript',
+  ],
+  plugins: ['jest', '@typescript-eslint', 'import', 'prettier'],
```
Comment (souyahia-monk) [inline configs/eslint-config-typescript-react/index.js:10]: do you really need to redefine all of those plugins here? Since they are already defined in @monkvision/eslint-config-typescript

File: apps/monk-test-app/package.json [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test --no-watch --no-watchAll",
+    "eject": "react-scripts eject"
```
Comment (souyahia-monk) [inline apps/monk-test-app/package.json:56]: you did not add a lint script ! please add a lint script (same as other projects)

File: .gitignore [hunk 12]
```diff
@@ -1,16 +1,44 @@
...
+
+# cache
+.eslintcache
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply .gitignore]: Done

File: .gitignore (hunk 1 above)
Comment (rohitsharma120582) [reply .gitignore]: Done

File: .gitignore (hunk 2 above)
Comment (rohitsharma120582) [reply .gitignore]: Done

File: apps/monk-test-app/.eslintignore (hunk 3 above)
Comment (rohitsharma120582) [reply apps/monk-test-app/.eslintignore]: Removed it...

File: apps/monk-test-app/.eslintignore [hunk 13]
```diff
@@ -0,0 +1,8 @@
+dist
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply apps/monk-test-app/.eslintignore:1]: Removed it...

File: apps/monk-test-app/.eslintignore (hunk 4 above)
Comment (rohitsharma120582) [reply apps/monk-test-app/.eslintignore]: Removed as no need to keep them

File: apps/monk-test-app/package.json [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-promise": "^6.1.1",
+    "eslint-plugin-react": "^7.32.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-utils": "^3.0.0",
+    "jest": "^29.3.1",
+    "mkdirp": "^1.0.4",
+    "prettier": "^2.7.1",
+    "regexpp": "^3.2.0",
+    "rimraf": "^3.0.2",
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply apps/monk-test-app/package.json:61]: Done

File: apps/monk-test-app/package.json (hunk 5 above)
Comment (rohitsharma120582) [reply apps/monk-test-app/package.json]: Done

File: apps/monk-test-app/package.json [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    "test": "react-scripts test --no-watch --no-watchAll",
+    "eject": "react-scripts eject"
+  },
+  "prettier": "@monkvision/prettier-config",
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply apps/monk-test-app/package.json]: Done

File: apps/monk-test-app/src/index.css (hunk 6 above)
Comment (rohitsharma120582) [reply apps/monk-test-app/src/index.css]: Done

File: apps/monk-test-app/tsconfig.json [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+      "esnext"
+    ],
+    "isolatedModules": true,
+    "jsx": "react-jsx",
+    "module": "esnext",
+    "noEmit": true,
+    "skipLibCheck": true
+  },
+  "include": ["src", "test"],
+  "exclude": ["node_modules", "**/*.spec.ts"]
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply apps/monk-test-app/tsconfig.json]: Done

File: configs/eslint-config-base/package.json (hunk 9 above)
Comment (rohitsharma120582) [reply configs/eslint-config-base/package.json]: Done

File: configs/eslint-config-typescript-react/index.js (hunk 10 above)
Comment (rohitsharma120582) [reply configs/eslint-config-typescript-react/index.js]: Done

File: configs/eslint-config-typescript-react/index.js [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+  settings: {
+    'import/resolver': {
+      node: {
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
+      },
+    },
+  },
+  rules: {
+    'prettier/prettier': ERROR,
+    'no-console': ERROR,
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply configs/eslint-config-typescript-react/index.js]: Done

File: apps/monk-test-app/package.json (hunk 11 above)
Comment (rohitsharma120582) [reply apps/monk-test-app/package.json]: Done

File: configs/eslint-config-base/package.json (hunk 8 above)
Comment (rohitsharma120582) [reply configs/eslint-config-base/package.json]: We have to downgrade the version here. As ESLint throws a mismatch error of peer dependencies while running the `yarn lint` command, fi go with higher version...

File: apps/monk-test-app/.eslintignore (hunk 13 above)
[the remark this answers is not in the document]
Comment (souyahia-monk) [reply apps/monk-test-app/.eslintignore:1]: I still see it in the file

File: apps/monk-test-app/package.json (hunk 14 above)
[the remark this answers is not in the document]
Comment (souyahia-monk) [reply apps/monk-test-app/package.json:61]: I still see them

File: .gitignore [hunk 18]
```diff
@@ -1,16 +1,40 @@
...
 
+# modules
 node_modules
 .idea
+
+# testing
+coverage/
+
+# production
+apps/monk-test-app/build/
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply .gitignore]: Done

**Review by deepakglobant — APPROVED** [review]

**Review by souyahia-monk — APPROVED** [review]
